src/intervals/intervals.jl

```diff
@@ -20,9 +20,10 @@ struct Interval{T<:Real} <: AbstractInterval{T}
     function Interval{T}(a::Real, b::Real) where T<:Real
 
         if validity_check
-
+            a = _normalisezero(a)
```

Comment:
I think `Interval` is by default with `validity_check` turned off? I wonder how often this would be effectively used

Reply:
Good catch! These lines should have been before the `if`... I'll push a new commit with that change.

Reply:
for example
```julia
julia> a = interval(0.0, -0.0)
[0, -0]

julia> sup(a)
-0.0
```

I wonder whether it would make more sense to have the check for +0 and -0 only in the `inf` `sup` functions and the normalisation only in the printing, what do you think?

Reply:
I think your observation does solve the corner case, which probably was due to `interval`. The one I was checking was `1 - (0..1)`. Before I was getting `[-0, 1]`; moving outside the `if` those lines I'm getting `[0, 1]`, but there is a (decorated) test not passing...

Reply:
Just pushed a new commit. It does take care of the corner case, or of your example, but there is a problem with one decorated test... Any idea?

Reply:
I think [this line](https://github.com/JuliaIntervals/IntervalArithmetic.jl/blob/master/src/decorations/functions.jl#L317) is the reason

Reply:
mmmm
```julia
julia> yy = DecoratedInterval(Interval(-0.0, 1.0), com)
[-0, 1]_com

julia> yyy = DecoratedInterval(Interval(0.0, 1.0), com)
[0, 1]_com

julia> xx = DecoratedInterval(Interval(-2.0, -0.1), com)
[-2, -0.1]_com

julia> atan(yyy, xx)
[1.67046, 3.1416]_com

julia> atan(yy, xx)
[1.67046, 3.1416]_dac
```

Reply:
I think that is a bug in atan? I don't think the behavior should differ whether it's +0 or -0

Reply:
Not sure yet... see [this line](https://github.com/JuliaIntervals/IntervalArithmetic.jl/blob/af22f4a1c24070de3ae25050526f222f57563cbe/src/decorations/functions.jl#L317). The sign seems to matter, and since we just suppressed it...

Reply:
in octave:
```octave
䰛ﮨ>> a = infsupdec(-0.0, 1.0)
a = [0, 1]_com
>> b = infsupdec(-2.0, -0.1)
b = [-2, -0.1]_com
>>
>> atan2(a, b)
ans = [1.6704, 3.1416]_dac
>> aa = infsupdec(0.0, 1.0)
aa = [0, 1]_com
>> atan2(aa, b)
ans = [1.6704, 3.1416]_dac
```

Reply:
I think the sign shouldn't matter for 0, that line is buggy I think

Reply:
@lbenet I pushed a commit that I think fixed it. Now it passes all tests (at least on my machine, let's see what the CI says) and behaves like the octave package. Before there were some discrepancies. I have to confess I am not fully sure why it works this way either. If you spot a mistake or would prefer to have this fix in a separate PR, I'd be glad to revert.

IntervalArithmetic.jl v0.20.1
```julia
(@v1.6) pkg> st IntervalArithmetic
      Status `~/.julia/environments/v1.6/Project.toml`
  [d1acc4aa] IntervalArithmetic v0.20.1

julia> atan(DecoratedInterval(-2..2), DecoratedInterval(-2.. -1))
[-3.1416, 3.1416]_dac

julia> atan(DecoratedInterval(-2..0), DecoratedInterval(-2.. -1))
[-3.1416, 3.1416]_dac

julia> atan(DecoratedInterval(0..0), DecoratedInterval(-2.. -1))
[3.14159, 3.1416]_com

julia> atan(DecoratedInterval(0..2), DecoratedInterval(-2.. -1))
[2.03444, 3.1416]_com
```

octave
```octave
>> atan2(infsupdec(-2, 2), infsupdec(-2, -1))
ans = [-3.1416, +3.1416]_def
>> atan2(infsupdec(-2, 0), infsupdec(-2, -1))
ans = [-3.1416, +3.1416]_def
>> atan2(infsupdec(0, 0), infsupdec(-2, -1))
ans = [3.1415, 3.1416]_dac
>> atan2(infsupdec(0, 2), infsupdec(-2, -1))
ans = [2.0344, 3.1416]_dac
```

after my latest commit
```julia
julia> atan(DecoratedInterval(-2..2), DecoratedInterval(-2.. -1))
[-3.1416, 3.1416]_def

julia> atan(DecoratedInterval(-2..0), DecoratedInterval(-2.. -1))
[-3.1416, 3.1416]_def

julia> atan(DecoratedInterval(0..0), DecoratedInterval(-2.. -1))
[3.14159, 3.1416]_dac

julia> atan(DecoratedInterval(0..2), DecoratedInterval(-2.. -1))
[2.03444, 3.1416]_dac

```